sopel/tools/target.py

```diff
@@ -107,9 +108,193 @@ def add_user(self, user, privs=0):
         """
         assert isinstance(user, User)
         self.users[user.nick] = user
-        self.privileges[user.nick] = privs
+        self.privileges[user.nick] = privs or 0
         user.channels[self.name] = self
 
+    def has_privilege(self, nick, privilege):
+        """Tell if a user has a ``privilege`` level or above in this channel.
+
+        :param str nick: a user's nick in this channel
+        :param int privilege: privilege level to check
+        :rtype: bool
+
+        This method checks the user's privilege level in this channel, i.e. if
+        it has this level or higher privileges::
+
+            >>> channel.add_user(some_user, plugin.OP)
+            >>> channel.has_privilege(some_user.nick, plugin.VOICE)
+            True
+
+        The ``nick`` argument can be either a :class:`str` or a
+        :class:`sopel.tools.Identifier`. If the user is not in this channel,
+        it will be considered has not having any privilege.
+
+        .. seealso::
+
+            There are other methods to check the exact privilege level of a
+            user, such as :meth:`is_oper`, :meth:`is_owner`, :meth:`is_admin`,
+            :meth:`is_op`, :meth:`is_halfop`, and :meth:`is_voiced`.
+
+        """
```

Comment:
This goes with the idea below (see line note in `is_oper()`).

```suggestion
        .. important::

            Not all IRC networks support all privilege levels. If you intend
            for your plugin to run on any network, it is safest to rely only
            on the presence of standard modes: ``+v`` (voice) and ``+o`` (op).

        """
```